src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -226,8 +226,16 @@ static bool IsContinuableState(WaitBehavior waitBehavior, CustomResourceSnapshot
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>A task.</returns>
     /// <remarks>
+    /// <para>
     /// This method returns a task that will complete with the resource is healthy. A resource
-    /// without <see cref="HealthCheckAnnotation"/> annotations will be considered healthy.
+    /// without <see cref="HealthCheckAnnotation"/> annotations will be considered healthy once
+    /// it reaches a <see cref="KnownResourceStates.Running"/> state.
+    /// </para>
+    /// <para>
+    /// If the resource enters an unavailable state such as <see cref="KnownResourceStates.FailedToStart"/> then
+    /// this method will continue to wait to enable scenarios where a resource could be restarted and recover. To
+    /// control this behavior use <see cref="WaitForResourceHealthyAsync(string, WaitBehavior, CancellationToken)"/>.
```

Comment:
> If the resource enters an unavailable state such as <see cref="KnownResourceStates.FailedToStart"/> then
> this method will continue to wait

The default behavior is controlled by the `ResourceNotificationServiceOptions.DefaultWaitBehavior`, which can be configured on the whole DistributedApplicationBuilder.

```suggestion
    /// If the resource enters an unavailable state such as <see cref="KnownResourceStates.FailedToStart"/> then
    /// this method will continue to wait to enable scenarios where a resource could be restarted and recover. To
    /// control this behavior use <see cref="WaitForResourceHealthyAsync(string, WaitBehavior, CancellationToken)"/>
    /// or configure the default behavior with <see cref="ResourceNotificationServiceOptions.DefaultWaitBehavior"/>.
```

Reply:
The current implementation of `WaitForResourceHealthyAsync` explicitly use `WaitOnResourceUnavailable` rather than `DefaultWaitBehavior` if no wait behaviour is set

https://github.com/dotnet/aspire/blob/423d678f9ca3fa489a7caba959d332fd2d24aab2/src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs#L232-L238

(I have proposed #7709 to change this)